docs/source/config_files.rst

```diff
@@ -32,64 +23,75 @@ Below is a list of supported configuration files (roughly in the order of build
-trying the other configuration files before deciding to use your own Dockerfile.
-
-With Dockerfiles, a regular Docker build will be performed.
-**If a Dockerfile is present, all other configuration files will be ignored.**
-
-See the `Binder Documentation <https://mybinder.readthedocs.io/en/latest/dockerfile.html>`_ for
-best-practices with Dockerfiles.
-
 .. _environment-yml:
 
-``environment.yml``
-~~~~~~~~~~~~~~~~~~~
+``environment.yml`` - Install a Python environment
+==================================================
 
 ``environment.yml`` is the standard configuration file used by Anaconda, conda,
-and miniconda that lets you install Python packages.
-You can also install files from pip in your ``environment.yml`` as well.
-Our example `environment.yml <https://github.com/binder-examples/python-conda_pip/blob/master/environment.yml>`_
-shows how one can specify a conda environment for repo2docker.
+and miniconda that lets you install packages in the data analytics stack (it
+primarily installs Python packages, though can be used to install a range of
+non-Python packages as well).
+
+.. note::
+
+   You can install files from pip in your ``environment.yml`` as well.
+   For example, see the `binder-examples environment.yml
+   <https://github.com/binder-examples/python-conda_pip/blob/master/environment.yml>`_
+   file.
 
 You can also specify which Python version to install in your built environment
 with ``environment.yml``. By default, ``repo2docker`` **installs
 Python 3.6** with your ``environment.yml`` unless you include the version of
 Python in the file.  ``conda`` supports Python versions 3.6, 3.5, 3.4, and 2.7.
-``repo2docker`` support is best with Python 3.6, 3.5, and 2.7. If you include
-a Python version in a ``runtime.txt`` file in addition to your
-``environment.yml``, your ``runtime.txt`` **will be ignored**.
+``repo2docker`` support is best with Python 3.6, 3.5, and 2.7.
+
+.. warning::
+   If you include a Python version in a ``runtime.txt`` file in addition to your
+   ``environment.yml``, your ``runtime.txt`` **will be ignored**.
+
 
-``requirements.txt``
-~~~~~~~~~~~~~~~~~~~~
+``requirements.txt`` - Install a Python environment
+===================================================
 
 This specifies a list of Python packages that should be installed in your
 environment. Our
 `requirements.txt example <https://github.com/binder-examples/requirements/blob/master/requirements.txt>`_
 on GitHub shows a typical requirements file.
 
-``REQUIRE``
-~~~~~~~~~~~
+
+``setup.py`` - Install Python packages
+======================================
+
+To install your repository like a Python package, you may include a
+``setup.py`` file. repo2docker installs ``setup.py`` files by running
+``pip install -e .``.
+
+While one can specify dependencies in ``setup.py``,
+repo2docker **requires configuration files such as** ``environment.yml`` or
+``requirements.txt`` to install dependencies during the build process.
```

Comment:
This sentence was already here so maybe we should leave it along. However do you understand what it is trying to tell us? Dependencies from your `setup.py` will be installed by repo2docker and you don't need to have a `requirements.txt`/`environment.yml` :-/

Reply:
I.....have no idea what this means lol. Should I just remove this paragraph?